docs/src/introduction/scopes.md

```diff
@@ -0,0 +1,167 @@
+
+<!-- TODO: fact-check the statement on the behavior expectation of other scopes -->
+
+## Creating Scopes
+
+Scopes are only created when a function is called. When a function is called, a new scope is created for as long as the function is running. When the function is done, the scope is destroyed.
+
+## Scoping Rules
+
+### Isolation
+
+Child scopes are isolated from the parent scope. This means that variables defined in the child scope are not accessible from the parent scope and will be destroyed when the child scope is destroyed (unless they are referenced, see [closures](#closures)).
+
+```clojure
+'(fn 0 'a def) call
+
+a
+;; Throws an error because `a` is not defined in the main scope
+```
+
+### Access and Inheritance
+
+This newly created scope has access to the parent scope. It can read and write to the existing variables in the parent scope. However, if a variable is defined in a child scope, is will not be accessible from within the parent scope.
+
+In this sense, variables have full access to **existing** variables of the parent scope, but cannot introduce **new** variables into that scope.
+
+**Getting:**
+
+```clojure
+0 'a def
+
+'(fn a) call
+
+;; Pushes 0 to the stack
+;; [] -> [0]
+```
+
+**Setting:**
+
+```clojure
+0 'a def
+
+'(fn 1 'a set) call
+
+a
+;; Pushes 1 to the stack
+;; [] -> [1]
+```
+
+### Shadowing
+
+When a variable is defined in a child scope with the same name as a variable in the parent scope, the child scope's variable will "shadow" the parent scope's variable. This means that the child scope's variable will be used instead of the parent scope's variable.
+
+To the function, it will have access to its own variable while the parent scope's variable is still accessible from within the parent scope.
+
+```clojure
+0 'a def
+
+'(fn 1 'a def a) call
+;; Pushes 1 to the stack
+;; [] -> [1]
+
+a
+;; Pushes 0 to the stack
+;; [] -> [0]
+```
+
+### Closures
+
+When a variable is referenced in a child scope, it will be kept alive in the parent scope. This is called a closure. This is a powerful feature that allows for more dynamic programming.
+
+```clojure
+'(fn
+  0 'a def
+
+  '(fn a)
+)
+
+;; Call the parent function
+;; (returns the child function)
+call
+
+;; Call the child function
+call
+;; Pushes 0 to the stack
+;; [] -> [0]
+```
+
+As you can see, the child function still has access to the parent scope's variable `a` even though the parent function finished executing.
+
+#### Implementation
+
+Closures are implemented by a scanning system. The scanner is triggered whenever a function is pushed to the stack and recursively scans all symbols that the function contains. However, the scanner is dumb and doesn't know how a function will use its symbols. Because of this, the scanner makes decisions based on whether a symbol in the function exists in the parent scope.
```

Comment:
> *'Closures are implemented by a scanning system...'*

Maybe something such as, *'Closures are enabled by the scanning system...'*, to be a little clearer.